pkg/kyverno/common/common.go

```diff
@@ -712,6 +733,51 @@ func GetResourceAccordingToResourcePath(fs billy.Filesystem, resourcePaths []str
 	return resources, err
 }
 
+func ProcessImageVerifyEngineResponse(policy *v1.ClusterPolicy, imageVerifyResponse *response.EngineResponse, resPath string, rc *ResultCounts) error {
+	printCount := 0
+	for _, rule := range policy.Spec.Rules {
+		if !rule.HasVerifyImages() {
+			continue
```

Comment:
This for loop doesn't seem to be doing anything, is it missing an invariant?

Reply:
There may be chance when mixed type of policy (policy containing 2 different type of rule) tested using `test` command it may lead to give improper result as it was occurring in issue #2934. Currently, I cannot test it because `test` command for `imageVerify` rule doesn't support.

Reply:
@viveksahu26 maybe I'm missing something but this for loop seems to be going over the rules in a policy and executing continue for the inner loop that isn't doing anything else. Here's a simple reproduction of what I mean https://go.dev/play/p/63zoqCtIwOG

I'm guessing what you wanted to do was move this continue statement a few lines below in the next for loop.

Reply:
for _, rule := range policy.Spec.Rules {
		if !rule.HasVerifyImages() {
			continue
	       }else{
	              break
	       }
}

The reason to continue is not to involve rules other than `imageVerify` rules. Is above loop looks good now ?

Reply:
Oh, It was my mistake that I was not able to catch your point @samj1912 . Now, I got it what you wanted to say.